packages/store-shared/src/validation/AjvEventsValidator.ts

```diff
@@ -0,0 +1,35 @@
+import { Event } from "@withorbit/core";
+import { EventsValidator, EventsValidatorError } from "./eventsValidator";
+import Ajv, { ValidateFunction as AjvValidationFunction } from "ajv";
+import DatabaseEventsSchema from "./events.json";
+
+export class AjvEventsValidator implements EventsValidator {
+  private readonly validator: AjvValidationFunction;
+
+  constructor() {
+    const ajv = new Ajv({
+      allowUnionTypes: true,
+      verbose: true,
+    });
+    this.validator = ajv.compile(DatabaseEventsSchema);
+  }
+
+  validateEvents(events: Event[]): EventsValidatorError | true {
+    const isValidTypes = this.validator(events);
+    if (isValidTypes !== true) {
+      if (this.validator.errors) {
+        return {
+          errors: this.validator.errors.map((error) => ({
+            message: `${error.schemaPath} ${error.message}`,
+          })),
+        };
+      } else {
+        return {
+          errors: [{ message: "An unknown validation error occurred" }],
+        };
+      }
+    }
+    // TODO: validate the existence of IDs
```

Comment:
I left this is as a todo, but would be happy to address it in this PR! I just wasn't sure which IDs would make sense to have their existence validated? Right now, if you try to reschedule before the initial ingest, it will fail. What other cases should we protect against?

Reply:
Yeah, this is a key limitation. Some portion of the validity of events is contingent on the state of the entity it's being applied to.

But I think that's OK: the `Database` will still fail `putEvents` when trying to apply the event to the entity.